### PR TITLE
Ajout de fonctionnalités

### PR DESCRIPTION
**Pull request by jmjoary**
- La requête de suppression n'était pas authentifiée
- Permet l'utilisation du filtre isNotNull
- Ajout de propriétés calculées pour la classe DirectusUser
- Ajout de filtre pour la recherche d'utilisateur
- Ajout d'une variable field pour choisir les champs
- Ajout des tests unitaires liés

### CODE REVIEW

File: lib/src/model/directus_user.dart [hunk 1]
```diff
@@ -43,4 +43,19 @@ class DirectusUser {
 
   DirectusUser.from({required this.id, required String email})
       : allProperties = {"id": id, "email": email};
+
+  String get fullName {
+    final String currentFirstName = (firstname == null) ? "" : firstname!;
```
Comment (maxbritto) [inline lib/src/model/directus_user.dart:48]: I'd rather avoid using forced unwrap (usage of ! for null values). It might lead to crashes in multithreaded contexts.
You can use a local variable to test and unwrap safely : 
```dart
final String currentFirstName = firstname;
return currentFirstName ?? "";
```

File: test/model/directus_user_test.dart [hunk 2]
```diff
@@ -36,5 +36,33 @@ main() {
       expect(sut.firstname, null);
       expect(sut.lastname, null);
     });
+
+    test('Reading first, last and full name', () {
```
Comment (maxbritto) [inline test/model/directus_user_test.dart:40]: Could you add tests for edge cases (firstname and/or lastname are null for example) ? Thanks!

**Review by maxbritto — CHANGES_REQUESTED** [review]
Thanks for this pull request. 
I've just noticed 2 small edits that I commented bellow.
Could you make those changes and resubmit so I can merge it into the project ?
Thanks!

**Review by maxbritto — APPROVED** [review]
Perfect, thanks!
Merging now 👍
